gui/src/components/Balances.tsx

```diff
@@ -1,22 +1,38 @@
 import { Stack, Typography } from "@mui/material";
+import { invoke } from "@tauri-apps/api/tauri";
 import { erc20ABI } from "@wagmi/core";
+import { useEffect } from "react";
 import { formatUnits } from "viem";
 import { useBalance, useContractRead } from "wagmi";
 
 import { useAccount } from "../hooks";
 import { useInvoke } from "../hooks/tauri";
+import { useNetworks } from "../hooks/useNetworks";
 import { useRefreshTransactions } from "../hooks/useRefreshTransactions";
 import { Address } from "../types";
 import { CopyToClipboard } from "./CopyToClipboard";
 import Panel from "./Panel";
 
 export function Balances() {
   const address = useAccount();
+  const { currentNetwork } = useNetworks();
   const { data: balances, mutate } = useInvoke<[Address, string][]>(
     "db_get_erc20_balances",
     { address }
   );
 
+  useEffect(() => {
+    (async () => {
+      if (currentNetwork && address) {
+        await invoke("alchemy_fetch_balances", {
+          chainId: currentNetwork.chain_id,
+          address,
+        });
+        mutate();
+      }
+    })();
+  }, [address, currentNetwork, mutate]);
```

Comment:
let's keep this file out of this PR please.
I'm also not sure if this is the best strategy to fetch, since it will trigger every time the UI loads